### PR TITLE
Bug 1383500 - Allow giving a default product in the URL in new-bug

### PR DESCRIPTION
**Pull request by SebastinSanty**

### CODE REVIEW

File: new_bug.cgi [hunk 1]
```diff
@@ -37,6 +37,8 @@ use Bugzilla::Field;
 use Bugzilla::Status;
 use Bugzilla::UserAgent;
 use Bugzilla::BugMail;
+use Data::Dumper;
```
Comment (floatingatoll) [inline new_bug.cgi]: Consider removing this.

File: new_bug.cgi [hunk 2]
```diff
@@ -46,6 +48,8 @@ my $cgi      = Bugzilla->cgi;
 my $template = Bugzilla->template;
 my $vars     = {};
 
+warn Dumper($cgi->path_info);
```
Comment (floatingatoll) [inline new_bug.cgi]: Definitely remove this.

File: template/en/default/global/header.html.tmpl [hunk 3]
```diff
@@ -94,6 +94,7 @@
                       "http://www.w3.org/TR/html4/loose.dtd">
 <html lang="en">
   <head>
+    [% IF use_base %] <base href="[% urlbase FILTER html %]"> [% END %]
```
Comment (floatingatoll) [inline template/en/default/global/header.html.tmpl]: How is this change related to accepting a default product?

File: .htaccess [hunk 4]
```diff
@@ -53,7 +53,7 @@ RewriteRule ^__version__$ version.json [L]
 RewriteRule ^__heartbeat__$ heartbeat.cgi [L]
 
 # New single page interface for filing bugs
-RewriteRule ^new[-_]bug$ new_bug.cgi [L,QSA]
+RewriteRule ^new[-_]bug($|/) new_bug.cgi [L,QSA]
```
Comment (floatingatoll) [inline .htaccess]: Is this one single line change the absolute minimum change necessary to repair bug 1383500?

Comment (SebastinSanty) [inline .htaccess]: Yes, to accept other extended URLs. Plan is to have something like `new-bug/Core` file a bug in Core.

File: new_bug.cgi (hunk 2 above)
Comment (SebastinSanty) [inline new_bug.cgi]: Yes, sure. This is in debugging stage. I should've marked the PR as WIP.

**Comment by floatingatoll** [comment]
Marked as WIP, apologies for the midflight review!

File: new_bug.cgi (hunk 1 above)
Comment (dylanwh) [inline new_bug.cgi]: Remove this please

File: template/en/default/global/header.html.tmpl (hunk 3 above)
Comment (dylanwh) [inline template/en/default/global/header.html.tmpl]: we can remove this since we're using window.location.hash

File: template/en/default/bug/new_bug.html.tmpl [hunk 5]
```diff
@@ -14,7 +14,9 @@
     javascript_urls = ['extensions/BugModal/web/new_bug.js']
     style_urls = ['extensions/BugModal/web/new_bug.css']
 %]
-[% PROCESS global/header.html.tmpl %]
+[% PROCESS global/header.html.tmpl
+    use_base = 1
```
Comment (dylanwh) [inline template/en/default/bug/new_bug.html.tmpl]: don't need this now, although leaving the option is useful

File: extensions/BugModal/web/new_bug.js [hunk 6]
```diff
@@ -1,30 +1,78 @@
...
+            });
+        },
+        function() {
+            alert("Network issues. Please refresh the page and try again");
+        }
+    );  
+}
 
 $(document).ready(function() {
+    var current_hash = window.location.hash;
```
Comment (dylanwh) [inline extensions/BugModal/web/new_bug.js]: perhaps 

```javascript
var product_name = window.location.hash ? window.location.hash.substr(1).toLowerCase 
                                                                       : null;
```

File: extensions/BugModal/web/new_bug.js [hunk 7]
```diff
@@ -1,30 +1,78 @@
...
     bugzilla_ajax(
             {
                 url: 'rest/bug_modal/initial_field_values'
             },
             function(data) {
                 initial = data
+                if (current_hash) {
+                    for (product in initial.products) {
+                        if (initial.products[product].name === current_hash.substr(1)) {
```
Comment (dylanwh) [inline extensions/BugModal/web/new_bug.js]:
```javascript
if (initial.products[product].name.toLowerCase() === product_name_lc)
```

File: extensions/BugModal/web/new_bug.js [hunk 8]
```diff
@@ -1,30 +1,78 @@
...
+                            product_name = current_hash.substr(1);
+                            var product_node = document.createElement("input");
+                            $("#product_wrap").html('<input name="product" type="hidden" id="product" value="' + product_name + '"><h3 style="padding-left:20px;">'+ product_name +'</h3>');
```
Comment (dylanwh) [inline extensions/BugModal/web/new_bug.js]: This could lead to XSS (maybe self-xss)

much safer:

```javascript
$("#product_wrap").html('<input name="product" type="hidden" id="product"><h3 id="product_name_heading"></h3>')
$("#product").value(product_name);
$("#product_name_heading").text(product_name);
```

File: extensions/BugModal/web/new_bug.js (hunk 6 above)
Comment (SebastinSanty) [inline extensions/BugModal/web/new_bug.js]: Wouldn't it be better, if we don't go with LowerCase? It is used by `component_load()` and `$("#product").value()` which interact with backend cgi directly. To get lower case hash in url covered, I can just put: 
```
if (initial.products[product].name.LowerCase() === current_hash.substr(1).LowerCase()) 
```

Comment (dylanwh) [inline extensions/BugModal/web/new_bug.js]: sounds good

**Comment by SebastinSanty** [comment]
@dylanwh Yes, that happens when the `new-bug` page is already loaded and then `#<product>` is appended to it. So should I add a manual reloading (on appending url with hash) in the script to avoid this confusion? How do you expect this to be?

**Comment by dylanwh** [comment]
@SebastinSanty I think it should possible to update when the URL changes, which I've just been looking into.

File: extensions/BugModal/web/new_bug.js [hunk 9]
```diff
@@ -121,6 +138,7 @@ $(document).ready(function() {
             document.getElementById('reset').style.display = "none";
             $("#description").prop('required',false);
         });
+<<<<<<< HEAD
```
Comment (dylanwh) [inline extensions/BugModal/web/new_bug.js]: remove the merge markers

**Comment by dylanwh** [comment]
\o/ #203 is in, so I've updated this against master. Let's hope the tests pass!
